### PR TITLE
Solve issue #2129 error message for bad headers format is unclear

### PR DESCRIPTION
**Pull request by alejogs4**
I have added validation to ensure that header being objects, and at the same time I tested this validation and I fixed a problem with one tests

### CODE REVIEW

**Comment by jasonsaayman** [comment]
Issue at: #2129

**Comment by roman40a** [comment]
Maybe add to `index.d.ts` header should be Object or this structure `{[h: string]: string}`?

File: index.d.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export interface AxiosRequestConfig {
   baseURL?: string;
   transformRequest?: AxiosTransformer | AxiosTransformer[];
   transformResponse?: AxiosTransformer | AxiosTransformer[];
-  headers?: any;
+  headers?: object | undefined;
```
Comment (chinesedfan) [inline index.d.ts]: `headers?` includes `undefined`. And `object` in Typescript is a little wired, which means everything except primitive types, i.e. "let a: object = [];" is valid.

Personally, I suggest,

```
headers?: {[key: string]: string}
```

File: test/specs/requests.spec.js [hunk 2]
```diff
@@ -33,6 +33,18 @@ describe('requests', function () {
     });
   });
 
+  it('should throw an error if given headers are not an object', function(done) {
```
Comment (chinesedfan) [inline test/specs/requests.spec.js]: Adding the test here is fine. But in `test/specs/helpers/normalizeHeaderName.spec.js` would be better.

File: tscommand-57f971ce.tmp.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+test\typescript\axios.ts --sourcemap --removeComments --lib es5,es2015.promise,dom --target ES5
```
Comment (chinesedfan) [inline tscommand-57f971ce.tmp.txt]: Delete the unnecessary file.

File: test/unit/adapters/http.js [hunk 4]
```diff
@@ -313,6 +313,7 @@ describe('supports http with nodejs', function () {
         assert.fail();
       }).catch(function (err) {
         assert.equal(err.message, 'ENOENT: no such file or directory, open \'/does/not/exist\'');
+      }).finally(function() {
```
Comment (chinesedfan) [inline test/unit/adapters/http.js]: Would you mind explaining why add this line?

**Comment by jasonsaayman** [comment]
Hi,

Thank you for this pull request, however it seem that it has gone stale and is either far out of date with the current master head or has unresolved review issues. Should you feel that this pull request is still relevant please submit a new pull request.

Thanks

**Pull request closed without merging**